pkg/healthcheck/healthcheck.go

```diff
@@ -1594,6 +1600,31 @@ func (hc *HealthChecker) RunChecks(observer CheckObserver) (bool, bool) {
 	return success, warning
 }
 
+func (hc *HealthChecker) RunChecksExitOnError() (bool, bool) {
```

Comment:
nit: again on naming, we could perhaps say `RunWithExitOnError`. Checks might be implied since it's a method on the health checker. Again, completely fine not to make the change, whatever sounds best to you.